Build fails on Mac OS X after commit 96c3756

The current master branch (commit 08bb806) fails to build on Mac OS X with the following error message:

```
parse_regressor.cc:40:24: error: no matching function for call to 'merand48'
{*iter = (float)(0.1 * merand48(ind));
./rand48.h:3:7: note: candidate function not viable: no known conversion from 'size_t' (aka 'unsigned long') to 'uint64_t &' (aka 'unsigned long long &') for 1st argument
float merand48(uint64_t& initial);
```

I noticed that a _size_t_ is passed on to _merand48_ instead of _uint64_t_, should be an easy fix. I think this maybe a compiler dependant conversion issue. Build is successful for commit (c593f3a)

--- a/vowpalwabbit/array_parameters.h
+++ b/vowpalwabbit/array_parameters.h
@@ -137,7 +137,7 @@ public:
 	      T(iter);
 	  }
 	
-	template<void(*T)(iterator&, size_t)> //for random initialization of weights (with stride) 
+	template<void(*T)(iterator&, uint64_t)> //for random initialization of weights (with stride) 
 	inline void set_default()
 	{  uint32_t stride = 1 << _stride_shift;
 	   iterator iter = begin();
@@ -145,7 +145,7 @@ public:
 			T(iter, i);
 	}
 
-	template<void(*T)(iterator&, size_t, uint32_t)> //for random initialization of the entire weight_vector 
+	template<void(*T)(iterator&, uint64_t, uint32_t)> //for random initialization of the entire weight_vector 
 	inline void set_default()
 	{ uint32_t stride = 1 << _stride_shift;
 	iterator iter = begin();

--- a/vowpalwabbit/gd_mf.cc
+++ b/vowpalwabbit/gd_mf.cc
@@ -192,7 +192,7 @@ void mf_train(gdmf& d, example& ec)
     THROW("cannot use triples in matrix factorization");
 }
 
-void set_rand(weight_parameters::iterator& iter, size_t index, uint32_t stride)
+void set_rand(weight_parameters::iterator& iter, uint64_t index, uint32_t stride)
 { 
 	for (weights_iterator_iterator<weight> w = iter.begin(); w != iter.end(stride); ++w, ++index)
 	  *w = (float)(0.1 * merand48(index));

--- a/vowpalwabbit/lda_core.cc
+++ b/vowpalwabbit/lda_core.cc
@@ -667,7 +667,7 @@ public:
 	initial_weights(weight initial, weight initial_random, bool random, uint32_t lda ) 
 		: _initial(initial), _initial_random(initial_random), _random(random), _lda(lda)
 	{}
-	void operator()(weight_parameters::iterator& iter, size_t index)
+	void operator()(weight_parameters::iterator& iter, uint64_t index)
 	{ if (_random)
 	  { for (weights_iterator_iterator<weight> k = iter.begin(); k != iter.end(_lda); ++k, ++index)
 		{  *k = (float)(-log(merand48(index) + 1e-6) + 1.0f);

